doc/development/guide/architecture.rst

```diff
@@ -0,0 +1,195 @@
+.. role:: html(raw)
+   :format: html
+
+Architectural overview
+======================
+
+PennyLane allows optimization and machine learning of quantum and hybrid
```

Comment:
Is it correct to say that `PennyLane allows machine learning of computations`? I would probably reformulate this sentence. :thinking:

Reply:
That's a fair point! I've reformulated it